src/XrdCrypto/XrdCryptosslgsiAux.cc

```diff
@@ -656,17 +656,61 @@ int XrdCryptosslX509CreateProxyReq(XrdCryptoX509 *xcpi,
       PRINT("EEC certificate has expired");
       return -kErrPX_ExpiredEEC;
    }
+
+   // These will be assigned dynamically allocated ssl structures later.
+   // An RAII style guard is used to free them on exit from this method.
+   EVP_PKEY       *ekro  = 0;
```

Comment:
If you don't want the bother of an entire class, you could do something like:

```
std::unique_ptr<EVP_PKEY, decltype(EVP_PKEY_free)> ekro(nullptr, &EVP_PKEY_free)
```

Type definition is uglier but maybe clearer to some?